### PR TITLE
Feature 6 login

### PR DESCRIPTION
**Pull request by claregillis**
For this feature, I did my best to follow clean architecture, but the depencency inversion might be a bit off. I added all of the necessary classes and documentation for Login, as well as tests for most things. You'll see that here and in feature-9-review there are a couple of identical classes (GetUser, UserExists) that I think we can combine later, as long as it doesn't interfere with single-responsibility. 

So far, LoginDatabaseController just has a dummy method that returns a set user for a specific username and an empty user otherwise; that's just for testing. There's a TODO to implement that method later where it will return the user from the database or an empty user.

I still need to implement some stuff on the LoginPage because it will need Swing.

LMK if you have any questions/comments/concerns

### CODE REVIEW

File: src/main/java/forms/LoginForm.java [hunk 1]
```diff
@@ -1,4 +1,66 @@
...
+        this.username = username;
+        this.password = password;
+    }
+
+    public
+
+    @Override
+    boolean validateForm() {
+        return password.length() > 0 && username.length() > 0;
+    }
```
Comment (lukecheseldine) [inline src/main/java/forms/LoginForm.java:40]: reformat code pls

**Review by lukecheseldine — APPROVED** [review]
absolutely cracked. please reformat code though :)

File: src/main/java/forms/LoginForm.java (hunk 1 above)
Comment (claregillis) [reply src/main/java/forms/LoginForm.java:38]: fixed! thanks for noticing

**Review by ethanrstark — DISMISSED** [review]

**Review by lukecheseldine — APPROVED** [review]
still cracked, pls teach me your ways

**Review by danielqiuu — APPROVED** [review]

**Review by lukecheseldine — APPROVED** [review]
noice

**Review by ethanrstark — DISMISSED** [review]

**Review by danielqiuu — APPROVED** [review]

**Review by liu-samuel — APPROVED** [review]

**Review by helenastra — APPROVED** [review]

**Review by ericguo1230 — APPROVED** [review]

**Review by ethanrstark — APPROVED** [review]

**Pull request merged**